### PR TITLE
Fix blog ordering on blog index page

### LINKED ISSUE #97: BlogPosts published on the same day seem to be out of order
# Steps to Reproduce

1. Publish a test blog post.
2. Publish a second test blog post, with the same **Publication date** as the first post.
3. View the blog post index (`/blog`)

## Expected behavior

The second blog post, which was published after the first, should appear above the first blog post in the list of blog posts, since blogs typically display posts sorted from first to last posted.

## Observed Behavior

The first blog post appears above the second blog post.

# Additional Notes

This happens because we only store and sort by BlogPost `date`, which is a `DateField`, not a `DateTimeField`. There are at least two ways to fix this:

1. Change the `BlogPost.date` DateField to a `BlogPost.published` DateTimeField. This might make entering the publication date a little more annoying in the admin, I'm not sure if the admin widget has ergonomic defaults.
2. Additionally sort by another factor in addition to `BlogPost.date`, e.g. leverage the auto-incrementing primary key.

### PR DESCRIPTION
**Pull request by redshiftzero**
Fixes #97 by ordering blog posts on the index page by the primary key column `id` instead of the `date` column as suggested by @garrettr in the issue description. 

Adds a unit test to verify this behavior and prevent future regressions.

### CODE REVIEW

File: securethenews/blog/models.py [hunk 1]
```diff
@@ -56,6 +56,6 @@ def posts(self):
         posts = BlogPost.objects.live().descendant_of(self)
 
         # Order by most recent date first
-        posts = posts.order_by('-date')
+        posts = posts.order_by('-id')
```
Comment (garrettr) [inline securethenews/blog/models.py]: This is a good idea for achieving a fix with minimal changes, but I can think of a potential problem. Imagine:

1. Create "Blog Post 1", save as draft. It gets an `id` of 1 (because we are using the default behavior of auto-incrementing primary keys).
2. Create "Blog Post 2", save as draft. It gets an `id` of 2.
3. Publish "Blog Post 2". It is shown first in the list, since it is the only entry in the list.
4. Publish "Blog Post 1". Now it is shown first in the list, _even though it was published second_. This is a bug.

The problem is that the primary key is created when the object is first saved, which may not be the same as when it is first published (since it may be saved as a draft, and other posts created and/or published, before it is finally published).

File: securethenews/blog/tests.py [hunk 2]
```diff
@@ -1,3 +1,42 @@
...
+            blog_index_page.add_child(instance=blog_post)
+
+
+    def test_ordering_of_same_day_blogs_on_index(self):
+        """Verify that blog posts posted on the same day are ordered with the most recent at the top of the page."""
+
+        blog_index = BlogIndexPage.objects.first()
+
+        self.assertEqual(blog_index.posts[0].title, 'Second Blog Post')
+        self.assertEqual(blog_index.posts[1].title, 'First Blog Post')
```
Comment (garrettr) [inline securethenews/blog/tests.py]: This test is :100:. Very clear, and beautifully written. Nice job!

File: securethenews/blog/models.py (hunk 1 above)
Comment (conorsch) [inline securethenews/blog/models.py]: That's a good description of a real edge case, @garrettr, but the proposed changes are generally applicable enough that this makes a good candidate for merge. Especially with the tests! 

Given the low rate of publishing on the site, I think we can live with the edge case, especially since we're totally aware of it. If it becomes an issue we can use solution 1) that you described in #97 and write a migration for it.

Comment (garrettr) [inline securethenews/blog/models.py]: @conorsch I hear you, but I feel pretty strongly about _not_ merging code that has known bugs. This is also such an easy case to encounter that I wouldn't describe it as an "edge" case.

This PR is still super worthwhile because the test is spot-on and should be kept as-is (well, plus an additional test to test the bug I described at the start of this thread). However, I think that the best fix is the first option I suggested in #97 (explicit is better than implicit!). That fix should be easy to implement and I'm happy to help @redshiftzero out with writing the migration (the only even slightly tricky part).

Comment (garrettr) [inline securethenews/blog/models.py]: Ok, so I just tested this out and the "potential problem" that I had originally described does not appear to be a problem after all! I'm not sure why. I guess Wagtail reassigns the primary key when a page is published? Anyway, this fix and test look good after all, so I'm going to ahead and merge this. Sorry for holding up the merge on what turned out to be a non-issue!
